File: problems/Entrenamiento UPC/Problemset/B/submissions/accepted/solucion.py
from math import factorial
from collections import Counter

def count_perms(s):
    freq = Counter(s)
    n = len(s)
    denom = 1
    for v in freq.values():
        denom *= factorial(v)
    return factorial(n) // denom

t = int(input())
for _ in range(t):
    n = int(input())
    s = input()
    
    min_perms = -1
    best = s
    
    # Probar todas las operaciones posibles (n <= 10, O(n^2) es viable)
    for i in range(n):
        for j in range(n):
            temp = list(s)
            temp[i] = s[j]
            temp_str = ''.join(temp)
            p = count_perms(temp_str)
            if min_perms == -1 or p < min_perms or (p == min_perms and temp_str < best):
                min_perms = p
                best = temp_str
    
    print(best)

File: problems/Entrenamiento UPC/Problemset/B/submissions/accepted/solucion.cpp
#include <iostream>
#include <string>
#include <map>
#include <set>
#include <algorithm>
using namespace std;

// Calcular numero de permutaciones distintas
long long countPerms(const string& s) {
    int n = s.size();
    map<char, int> freq;
    for (char c : s) freq[c]++;
    
    long long num = 1;
    for (int i = 1; i <= n; i++) num *= i;
    for (auto& p : freq) {
        long long d = 1;
        for (int i = 1; i <= p.second; i++) d *= i;
        num /= d;
    }
    return num;
}

int main() {
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;
        string s;
        cin >> s;
        
        long long minPerms = -1;
        string best = s;
        
        // Probar todas las operaciones posibles
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                string temp = s;
                temp[i] = s[j];
                long long p = countPerms(temp);
                if (minPerms == -1 || p < minPerms || (p == minPerms && temp < best)) {
                    minPerms = p;
                    best = temp;
                }
            }
        }
        
        cout << best << endl;
    }
    return 0;
}
